### PR TITLE
feat: add GurbaniNow API to fetch data for shabads and search results

### LINKED ISSUE #125: Connect real data to barebones v1
For #95 and #124, we need to add a data layer and functionality.

Drop in SOS SQLite, query it raw, replace later on when v5 is out and compatible.

### Acceptance Criteria

- Rudimentary querying with LIKE operator. No need for smart ordering like on presenter at this point.
- Bookmarks listing, supplemented by the collections json @bhajneet will supply
- Shabad view should render data

### PR DESCRIPTION
**Pull request by Harjot1Singh**
### Summary of PR
Adds live data from GurbaniNow API. Supports banis and shabads. Links together search results to opening a new shabad. Adds mechanism for loading bani into gurbani screen, making it trivial to integrate #126. 

### Tests for unexpected behavior
- AuToMaGiC TeSt SuItE

### Time spent on PR
4 hours

### Linked issues
Fix #125 

### Reviewers
@saihaj @bhajneet @AkalUstat

### CODE REVIEW

File: src/types/utils.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+export type Override<T1, T2> = Omit<T1, keyof T2> & T2
```
Comment (saihaj) [inline src/types/utils.ts:1]: Maybe we should comment this?

**Review by bhajneet — APPROVED** [review]

File: src/screens/Gurbani/index.tsx [hunk 2]
```diff
@@ -1,23 +1,37 @@
...
 import Lines from './Lines'
 import Navbar from './Navbar'
 
-type ShabadQuery = { queryKey: [string] }
-const shabadQuery = ( { queryKey }: ShabadQuery ) => getShabad( ...queryKey )
+type Loaders = {
+  [screen in ContentTypes]: ( options : { queryKey: [string] } ) => Promise<{ lines: LineData[] }>
+}
+
+// Loaders all return a common interface. Is there a better way to deal with specifics of each type?
```
Comment (saihaj) [inline src/screens/Gurbani/index.tsx:18]: In [Flow](https://flow.org/en/) they have [`$ObjMap`](https://flow.org/en/docs/types/utilities/#toc-objmap). In TypeScript there is not an exact same thing but something somewhat similar is achievable.

**Review by saihaj — APPROVED** [review]
🎉

File: src/data/gurbaninow/types.ts [hunk 3]
```diff
@@ -0,0 +1,78 @@
+type SourcePartial = {
```
Comment (AkalUstat) [inline src/data/gurbaninow/types.ts:1]: is this called partial because the values are optional?

**Review by AkalUstat — DISMISSED** [review]

File: src/data/gurbaninow/types.ts (hunk 3 above)
Comment (Harjot1Singh) [reply src/data/gurbaninow/types.ts:1]: No, because it's a reusable component of other types.

File: src/types/utils.ts (hunk 1 above)
Comment (Harjot1Singh) [reply src/types/utils.ts:1]: Can do

File: src/screens/Gurbani/index.tsx (hunk 2 above)
Comment (Harjot1Singh) [reply src/screens/Gurbani/index.tsx:18]: What's the difference between that and what I have now?

Comment (saihaj) [reply src/screens/Gurbani/index.tsx:18]: It is doing the same thing. I though you want a  template that is more generic and infer types

**Review by saihaj — APPROVED** [review]

**Pull request merged**